packages/editor/src/core/components/Cell/index.tsx

```diff
@@ -107,39 +109,51 @@ const Cell: React.FC<Props> = ({ nodeId, measureRef }) => {
 
   return (
     <div
-      ref={ref}
+      style={{
+        padding: `0 ${cellSpacing / 2}px`,
+        paddingBottom:
+          (!hasChildren || hasPlugin) && !inline ? `${cellSpacing}px` : null,
```

Comment:
hasChildren is a bit tricky. We unified layout and content plugins where a "layout" plugin is just a cell plugin that renders children.

Problem is that `hasChildren` does not tell you the difference between a plugin that can't have children and one that can have children but does not have any at the moment (e.g. an empty background). That was a slight oversight of mine.

this might lead to a padding jump here when you add the first children.

Reply:
hasChildren check here is done for the sake of this situation:

![image](https://user-images.githubusercontent.com/10179521/109724957-484f4500-7b65-11eb-87a3-7c58de9afb21.png)

The jumbo cell on the right has children (three rows), but it should not have its own vertical padding as it has been taken care of by the individual cells inside.

hasPlugin check simply makes sure we add padding to layout plugins, whether they have children or not.

![image](https://user-images.githubusercontent.com/10179521/109725688-5baee000-7b66-11eb-819e-3bc24408c03e.png)

The first background is empty (I left the internal padding on so it doesn't collapse) and will receive its gutter at the bottom. The second background has children (one cell), but being a standalone cell it should still receive its gutter which it does.

This logic doesn't really depend on whether we're nested (inside a layout plugin) or at first layout level.
The compensation is happening elsewhere.